convert2rhel/checks.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright(C) 2016 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+import logging
+
+from convert2rhel.pkghandler import call_yum_cmd
+
+logger = logging.getLogger(__name__)
+
+
+def check_repos_are_valid():
+    output, ret_code = call_yum_cmd(command="repolist", args="-v")
+    if "failure" in output or ret_code != 0:
+        logger.critical("Couldn't validate custom repositories. Please check custom repo config files.")
+    logger.debug("All custom repo appears to be valid.")
+
+
+def perform_pre_checks():
+    logger.task("Prepare: Checking if custom repos are valid")
```

Comment:
@fellipeh  the log message should be part of the specific check. this function should just call each *check function* in the file

Reply:
Agree with @pirat89, now I see that the only exception is when we're doing rollbacks. Otherwise all the logger.task are resides in main.py

Reply:
https://github.com/oamg/convert2rhel/search?q=loggerinst.task